### PR TITLE
Initial Release

### PR DESCRIPTION
**Pull request by moattarwork**
The PR includes the changes made to internal NoSql Project to make it ready for public use

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,15 +1,155 @@
...
+```
+
+### Usage
+s
+To use the repository, You can inject the repository to your class and all the relevant methods will be available to use. If you have selected WithTelemetry, ensure that the InstrumentationKey for your target application insights resource has been configured either in your calling application config: 
+
+```c#
+"ApplicationInsights": {
+    /* Run this command to get your Instrumentation key
+    az resource show --resource-group HoldingsFeedDevResourceGroup --name HoldingsFeedDevAppInsights --resource-type        "Microsoft.Insights/components" --query properties.InstrumentationKey
```
Comment (LewisPercival) [inline README.md]: should we have Holdings feed dev resource group in here if its to be open source?

**Review by davidcassell — APPROVED** [review]
Looks good

**Review by LewisPercival — COMMENTED** [review]
one comment to be removed

File: src/Easify.NoSql.MongoDb/MongoDbRepository.cs [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+        }
+
+        public virtual Task<T> GetAsync(Expression<Func<T, bool>> condition,
+            Func<IFindFluent<T, T>, IFindFluent<T, T>> query,
+            CancellationToken cancellationToken)
+        {
+            return InternalGetAsync(condition, query, cancellationToken);
+        }
+
+        [Obsolete("Use function with IFindFluent interface")]
```
Comment (davidcassell) [inline src/Easify.NoSql.MongoDb/MongoDbRepository.cs]: As discussed, it'd be good not to have to bring Obsolete methods across

Comment (moattarwork) [reply src/Easify.NoSql.MongoDb/MongoDbRepository.cs]: I made some refactoring here but removed the Obsolete annotation as they were necessary

**Pull request merged**